sklearn/utils/validation.py

```diff
@@ -324,21 +324,26 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None, copy=Fal
     if isinstance(accept_sparse, str):
         accept_sparse = [accept_sparse]
 
+    numeric = dtype == "numeric"
+
     if sp.issparse(array):
-        if dtype == "numeric":
+        if numeric:
```

Comment:
I don't understand how the above can not pass. "numeric" is defined just above as "numeric", so the if will always pass.